course/constants.py

```diff
@@ -504,6 +504,7 @@ class grade_state_change_types:  # noqa
     report_sent = "report_sent"
     do_over = "do_over"
     exempt = "exempt"
+    session_reopened = "session_reopened"
```

Comment:
How is `session_reopened` different from a do-over?

Reply:
If `do_over` is the latest grade change, it will clear previous grades, with stringified state is `NONE`. If  `session_reopened` as just like `do_over`, but with stringified state `OTHER_STATE` when it is the last grade change.